File: js - client/includes/javascript/core/components/transform.js
import { Vector2 } from "./vector2.js"

export class Transform
{
    constructor()
    {

        this.position = new Vector2(0, 0)     // position in viewport units
        this.rotation = 0                     // rotation in degrees
        this.scale = new Vector2(1, 1)        
        this.zOrder = 0                       // lower = bottom, higher = top

    }

    /** Sets the x and y position of object in viewport units */
    SetPosition( x, y )
    {
        this.position.x = x
        this.position.y = y
    }

    /** Gets the objects position in viewport units */
    GetPosition()
    {
        return this.position
    }

    /** Sets the x and y scale of the object */
    SetScale( x, y )
    {
        this.scale.x = x
        this.scale.y = y
    }

    /** Gets the objects scale */
    GetScale()
    {
        return this.scale
    }

}

File: python - server/world_models/core/components/transform.py

class Transform:

    def __init__( self, position, rotation, scale):
        """
                            2D    or   3D
        :param position:    Vector2 or Vector3 position
        :param rotation:    float   or Vector3 rotation
        :param scale:       Vector2 or Vector3 scale
        """
        self.position = position
        self.rotation = rotation
        self.scale = scale
